docs/resources/dms_kafka_instance.md

```diff
@@ -0,0 +1,221 @@
+  password               = "Kafkatest@123"
+  vpc_id                 = data.huaweicloud_vpc.test.id
+  subnet_id              = data.huaweicloud_vpc_subnet.test.id
+  security_group_id      = huaweicloud_networking_secgroup.test.id
+  available_zones        = [data.huaweicloud_dms_az.test.id]
+  product_id             = data.huaweicloud_dms_product.test.id
+  engine_version         = data.huaweicloud_dms_product.test.version
+  specification          = data.huaweicloud_dms_product.test.bandwidth
+  partition_num          = data.huaweicloud_dms_product.test.partition_num
+  storage_space          = data.huaweicloud_dms_product.test.storage
+  storage_spec_code      = data.huaweicloud_dms_product.test.storage_spec_code
+  kafka_manager_user     = "kafka-user"
+  kafka_manager_password = "Kafkatest@123"
+
+  tags = {
+    key   = "value"
+    owner = "terraform"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) The region in which to create the DMS kafka instance resource.
+  If omitted, the provider-level region will be used. Changing this creates a new instance resource.
+
+* `name` - (Required, String) Specifies the name of the DMS kafka instance. An instance name starts with a letter,
+    consists of 4 to 64 characters, and supports only letters, digits, hyphens (-) and underscores (_).
+
+* `description` - (Optional, String) Specifies the description of the DMS kafka instance. It is a character
+    string containing not more than 1024 characters.
+
+* `engine_version` - (Required, String, ForceNew) Specifies the version of the kafka engine.
+    Valid values are "1.1.0" and "2.3.0".
+    Changing this creates a new instance resource.
+
+* `specification` - (Required, String, ForceNew) Specifies the baseline bandwidth of the DMS kafka instance,
+    that is, the maximum amount of data transferred per unit time. Unit: byte/s. Options: 100MB, 300MB, 600MB, 1200MB.
+    Changing this creates a new instance resource.
+
+* `storage_space` - (Required, Int, ForceNew) Specifies the message storage space. Value range:
+    - When specification is 100MB: 600–90000 GB
+    - When specification is 300MB: 1200–90000 GB
+    - When specification is 600MB: 2400–90000 GB
+    - When specification is 1200MB: 4800–90000 GB
+    
+    Changing this creates a new instance resource.
+
+* `storage_spec_code` - (Required, String, ForceNew) Specifies the storage I/O specification. Value range:
+    - When specification is 100MB: dms.physical.storage.high or dms.physical.storage.ultra
+    - When specification is 300MB: dms.physical.storage.high or dms.physical.storage.ultra
+    - When specification is 600MB: dms.physical.storage.ultra
+    - When specification is 1200MB: dms.physical.storage.ultra
+
+    Changing this creates a new instance resource.
+
+* `partition_num` - (Required, Int, ForceNew) Specifies the maximum number of topics in the DMS kafka instance.
```

Comment:
if the value depends on `specification`, we can use it as an internal parameter or set it as an attribute

Reply:
fixed